File: entoma-mqtt-sliders-clue.js
  var sensor;
  var value;
   
   // Accelerometer X
    var xccelerometer = document.getElementById("X-Accelerometer");
    var xvalue = document.getElementById("X-Accelerometervalue");
    xvalue.innerHTML = xccelerometer.value;
    
    xccelerometer.oninput = function() {
        xvalue.innerHTML = this.value;
        sensor = "slider/xaccelerometer";
        value = xccelerometer.value;
    }

    // 



    // Accelometer Y
    var yccelerometer = document.getElementById("Y-Accelerometer");
    var yvalue = document.getElementById("Y-Accelerometervalue");
    yvalue.innerHTML = yccelerometer.value;
    
    yccelerometer.oninput = function() {
        yvalue.innerHTML = this.value;
        sensor = "slider/yaccelerometer";
        value = yccelerometer.value;
    }

    // Accelometer Z

    var zccelerometer = document.getElementById("Z-Accelerometer");
    var zvalue = document.getElementById("Z-Accelerometervalue");
    zvalue.innerHTML = zccelerometer.value;
    
    zccelerometer.oninput = function() {
        zvalue.innerHTML = this.value;
        sensor = "slider/zccelerometer";
        value = zccelerometer.value;
    }

      //Gyroscope X
   var xgyroscope = document.getElementById("X-Gyroscope");
    var xgyroscopevalue = document.getElementById("X-Gyroscopevalue");
    xgyroscopevalue.innerHTML = xgyroscope.value;
    
    xgyroscope.oninput = function() {
        xgyroscopevalue.innerHTML = this.value;
        sensor = "slider/xgyroscope";
        value = xgyroscope.value;
    }

    // Gyroscope Y
    var ygyroscope = document.getElementById("Y-Gyroscope");
    var ygyroscopevalue = document.getElementById("Y-Gyroscopevalue");
    ygyroscopevalue.innerHTML = ygyroscope.value;
    
    ygyroscope.oninput = function() {
        ygyroscopevalue.innerHTML = this.value;
        sensor = "slider/ygyroscope";
        value = ygyroscope.value;
    }

    // Gyroscope Z

    var zgyroscope = document.getElementById("Z-Gyroscope");
    var zgyroscopevalue = document.getElementById("Z-Gyroscopevalue");
    zgyroscopevalue.innerHTML = zgyroscope.value;
    
    zgyroscope.oninput = function() {
        zgyroscopevalue.innerHTML = this.value;
        sensor = "slider/zgyroscope";
        value = zgyroscope.value;
    }

     //magnetic X
   var xmagnetic = document.getElementById("X-Magnetic");
    var xmagneticvalue = document.getElementById("X-Magneticvalue");
    xmagneticvalue.innerHTML = xmagnetic.value;
    
    xmagnetic.oninput = function() {
        xmagneticvalue.innerHTML = this.value;
        sensor = "slider/xmagnetic";
        value = xmagnetic.value;
    }

    // magnetic Y
    var ymagnetic = document.getElementById("Y-Magnetic");
    var ymagneticvalue = document.getElementById("Y-Magneticvalue");
    ymagneticvalue.innerHTML = ymagnetic.value;
    
    ymagnetic.oninput = function() {
        ymagneticvalue.innerHTML = this.value;
        sensor = "slider/ymagnetic";
        value = ymagnetic.value;
    }

    // magnetic Z

    var zmagnetic = document.getElementById("Z-Magnetic");
    var zmagneticvalue = document.getElementById("Z-Magneticvalue");
    zmagneticvalue.innerHTML = zmagnetic.value;
    
    zmagnetic.oninput = function() {
        zmagneticvalue.innerHTML = this.value;
        sensor = "slider/zmagnetic";
        value = zmagnetic.value;
    }

     // Pressure
   var pressure = document.getElementById("Pressure");
    var pressurevalue = document.getElementById("Pressurevalue");
    pressurevalue.innerHTML = pressure.value;
    
    pressure.oninput = function() {
        pressurevalue.innerHTML = this.value;
        sensor = "slider/pressure";
        value = pressure.value;
    }

     // Temperature
   var temperature = document.getElementById("Temperature");
    var temperaturevalue = document.getElementById("Temperaturevalue");
    temperaturevalue.innerHTML = temperature.value;
    
    temperature.oninput = function() {
        temperaturevalue.innerHTML = this.value;
        sensor = "slider/temperature";
        value = temperature.value;
    }

    // Humidity
    var humidity = document.getElementById("Humidity");
    var humidityvalue = document.getElementById("Humidityvalue");
    humidityvalue.innerHTML = humidity.value;
    
    humidity.oninput = function() {
        humidityvalue.innerHTML = this.value;
        sensor = "slider/humidity";
        value = humidity.value;
    }

    // red
    var red = document.getElementById("red");
    var redvalue = document.getElementById("redvalue");
    redvalue.innerHTML = red.value;
    
    red.oninput = function() {
        redvalue.innerHTML = this.value;
        sensor = "slider/red";
        value = red.value;
    }
    // green
    var green = document.getElementById("green");
    var greenvalue = document.getElementById("greenvalue");
    greenvalue.innerHTML = green.value;
    
    green.oninput = function() {
        greenvalue.innerHTML = this.value;
        sensor = "slider/green";
        value = green.value;
    }
    // blue
    var blue = document.getElementById("blue");
    var bluevalue = document.getElementById("bluevalue");
    bluevalue.innerHTML = blue.value;
    
    blue.oninput = function() {
        bluevalue.innerHTML = this.value;
        sensor = "slider/blue";
        value = blue.value;
    }

    // Proximity
    var proximity = document.getElementById("proximity");
    var proximityvalue = document.getElementById("proximityvalue");
    proximityvalue.innerHTML = proximity.value;
    
    proximity.oninput = function() {
        proximityvalue.innerHTML = this.value;
        sensor = "slider/proximity";
        value = proximity.value;
    }

    // Light Intensity
    var light = document.getElementById("light");
    var lightvalue = document.getElementById("lightvalue");
    lightvalue.innerHTML = light.value;
    
    light.oninput = function() {
        lightvalue.innerHTML = this.value;
        sensor = "slider/light";
        value = light.value;
    }


    function publishFunction() {
        console.log(sensor)
        console.log(value);
        client.publish(sensor, value)
    }

File: entoma-mqtt-sliders-clue.py
"""
To get started, check out the "Device Simulator Express: Getting Started" command in the command pallete, which you can access with `CMD + SHIFT + P` For Mac and `CTRL + SHIFT + P` for Windows and Linux.

To learn more about the CLUE and CircuitPython, check this link out:
https://learn.adafruit.com/adafruit-clue/circuitpython

Find example code for CPX on:
https://blog.adafruit.com/2020/02/12/three-fun-sensor-packed-projects-to-try-on-your-clue-adafruitlearningsystem-adafruit-circuitpython-adafruit/
"""

from adafruit_clue import clue
import paho.mqtt.client as mqtt

def display_text(temp):
    clue_data[0].text = "Accel: {} {} {} m/s^2".format(temp["slider/xccelerometer"] , temp["slider/yccelerometer"], temp["slider/zccelerometer"])
    clue_data[1].text = "Gyro: {} {} {} dps".format(temp["slider/xgyroscope"], temp["slider/ygyroscope"] , temp["slider/zgyroscope"])
    clue_data[2].text = "Magnetic: {} {} {} uTesla".format(temp["slider/xmagnetic"], temp["slider/ymagnetic"], temp["slider/zmagnetic"])
    clue_data[3].text = "Pressure: {} hPa".format(temp["slider/pressure"])
    clue_data[4].text = "Altitude: {:.0f} m".format(temp["slider/light"])
    clue_data[5].text = "Temperature: {} C".format(temp["slider/temperature"])
    clue_data[6].text = "Humidity: {} %".format(temp["slider/humidity"])
    clue_data[7].text = "Proximity: {}".format(temp["slider/proximity"])
    clue_data[8].text = "Color: R: {} G: {} B: {} C: {}".format(temp["slider/red"], temp["slider/green"], temp["slider/blue"], temp["slider/light"])
    clue_data.show()
sensor = {
    "slider/xccelerometer" :0,
    "slider/yccelerometer" : 0,
    "slider/zccelerometer" :0,
    "slider/xgyroscope" :0,
    "slider/ygyroscope" :0,
    "slider/zgyroscope" :0,
    "slider/xmagnetic" :0,
    "slider/ymagnetic" :0,
    "slider/zmagnetic" :0,
    "slider/pressure" :0,
    "slider/temperature" :0,
    "slider/humidity" :0,
    "slider/red" :0,
    "slider/green" :0,
    "slider/blue" :0,
    "slider/proximity" :0,
    "slider/light" :0,
}

# this function holds the condition  if rc is equal to 0 then the client will subscribe then after it will display the text
def on_connect(client, userdata, flags, rc):
    if rc == 0:
        client.subscribe("slider/#")
        display_text(sensor)

# this function  holds the messages or the payload
def on_message(client, userdata, msg):
    display_text(int(msg.payload.decode()))

clue.sea_level_pressure = 1020

clue_data = clue.simple_text_display(text_scale=2) 

client = mqtt.Client()
client.on_connect = on_connect #calling the function on_connect above
client.on_message = on_message #calling the function on_message above

client.connect("mqtt.eclipseprojects.io", 1883, 60) #connect to the browser

client.loop_forever() #loop forever